esphome/components/socket/lwip_raw_tcp_impl.cpp

```diff
@@ -178,26 +178,23 @@ class LWIPRawImpl : public Socket {
       errno = EINVAL;
       return -1;
     }
-    if (*addrlen < sizeof(struct sockaddr_in)) {
-      errno = EINVAL;
-      return -1;
-    }
-    struct sockaddr_in *addr = reinterpret_cast<struct sockaddr_in *>(name);
-    addr->sin_family = AF_INET;
-    *addrlen = addr->sin_len = sizeof(struct sockaddr_in);
-    addr->sin_port = pcb_->remote_port;
-    addr->sin_addr.s_addr = pcb_->remote_ip.addr;
+    this->ip2sockaddr_(&pcb_->local_ip, pcb_->local_port, name, addrlen);
     return 0;
```

Comment:
You should return what `ip2sockaddr_` returns:

```suggestion
    return this->ip2sockaddr_(&pcb_->local_ip, pcb_->local_port, name, addrlen);
```

Reply:
Yes, good catch, I forgot to change that when I moved the `addrlen` validation to `ip2sockaddr_`.